packages/core/src/builder.ts

```diff
@@ -614,4 +615,30 @@ export default class SchemaBuilder<Types extends SchemaTypes> {
       ? processedSchema
       : lexicographicSortSchema(processedSchema);
   }
+
+  withScalars<const ScalarRecord extends Record<string, GraphQLScalarType>>(scalars: ScalarRecord) {
+    const that = this as unknown as SchemaBuilder<
+      PothosSchemaTypes.ExtendDefaultTypes<
+        Types & {
+          Scalars: {
+            // Extract the Input and Output types from GraphQLScalarType's generics
+            [Property in keyof ScalarRecord]: ScalarRecord[Property] extends GraphQLScalarType<
+              infer TInternal,
+              infer TExternal
+            >
+              ? {
+                  Input: TInternal;
+                  Output: TExternal;
+                }
+              : never;
+          };
+        }
+      >
+    >;
+
+    for (const [name, scalar] of toPairs(scalars)) {
```

Comment:
Feel free to use Object.entries here with some type casts, not sure if adding toPairs and dealing with license stuff is worth it (up to you).